lib/trackerBlocking.js

```diff
@@ -66,11 +66,60 @@ const resourceTypes = new Set([
     'webtransport', 'webbundle', 'other'
 ])
 
+/**
+ * Unfortunately it's a little tricky to interact with TypeScript Enum types
+ * from JavaScript code, assigning the literal value (even if valid) results in
+ * a type coercion error. As a workaround, use the backtick syntax to get the
+ * possible enum values and then use those values to declare new types.
+ * Hopefully, interacting with Enums will get easier in the future and this can
+ * be removed.
+ * @typedef {`${chrome.declarativeNetRequest.DomainType}`} DomainType
+ * @typedef {`${chrome.declarativeNetRequest.RequestMethod}`} RequestMethod
+ * @typedef {`${chrome.declarativeNetRequest.ResourceType}`} ResourceType
+ * @typedef {`${chrome.declarativeNetRequest.RuleActionType}`} RuleActionType
+ * @typedef {Omit<chrome.declarativeNetRequest.RuleAction,
+ *                'type' | 'redirect' | 'requestHeaders' | 'responseHeaders'> &
+ *           {type: RuleActionType, redirect?: object,
+ *            requestHeaders?: object, responseHeaders?: object}} RuleAction
+ * @typedef {Omit<chrome.declarativeNetRequest.RuleCondition,
+ *               'domainType' | 'excludedRequestMethods' |
+ *               'excludedResourceTypes' | 'requestMethods' |
+ *               'resourceTypes'> &
+ *           {domainType?: DomainType, excludedRequestMethods?: RequestMethod[],
+ *            excludedResourceTypes?: ResourceType[],
+ *            requestMethods?: RequestMethod[],
+ *            resourceTypes?: ResourceType[]}} RuleCondition
+ * @typedef {Omit<chrome.declarativeNetRequest.Rule,
+ *                'id' | 'action' | 'condition'> &
+ *           {id?: number, action: RuleAction, condition: RuleCondition}} Rule
+ */
+
+/**
+ * @typedef {object} generateDNRRuleDetails
```

Comment:
Some of these are so long I'd make them helper classes which IMO makes them easier to read and also work with in future but not an issue at all :)

Reply:
Ack.